articles/password-checker-javascript/index.md

```diff
@@ -0,0 +1,252 @@
+
+<!-- CSS Styling -->
+<style>
+    .passwordInput{
+        margin-top: 5%; 
+        text-align :center;
+    }
+
+    .displayBadge{
+        margin-top: 5%; 
+        display: none; 
+        text-align :center;
+    }
+</style>
+    
+<body>
+    <div class="d-flex text-danger justify-content-center">
+        <ul style="list-style-type:none">
+            <li class="fs-4" style="margin-top: 10%; text-align :center">
+                Password Strength Checker
+            </li>
+            <li>
+                <input id="0101" class="form-control passwordInput" type="password" placeholder="Type your password" id="inputPassword">
+            </li>
+            <li>
+                <span id="0102" class="badge displayBadge">Weak</span>
+            </li>
+        </ul>
+    </div>     
+</body>
+</html>
+```
+
+- We will use Bootstrap 5 for styling. We will provide a CDN link for the bootstrap file and link it to the HTML file using the `<link>` tag. 
+
+- We write our custom styles using the `<style>` tag. We have two classes and their respective stylings.
+
+- In the `<body>` tag we have a `<div>` element.
+
+- In the `<div>`, we have an unordered list with the style `list-style-type:none` for removing bullets. It has three items (`<li>`):
+
+- The first is under class `fs-4` which makes the text size that of **h4**. We have also given it an inline styling.
+
+- The second item is an input where our password will be typed in. It is under the Bootstrap's class `form-control` and the class `passwordInput` which we custom styled in the `<style>` tag. The `form-control` shows the inputshould be used in a `form` element(We have not added a form since we are not sending any data).
+
+- The third item contains a `<span>` which is where the password strength will be labeled. It is under the class `displayBadge` which we custom styled and `badge` given by Bootstrap which makes it appear as a badge.
+
+#### JavaScript Code creation and explanation
+
+Create five variables: `timeout` for storing the timeout before a callback is called, `password` & `strengthBadge` for storing the input and span after traversing the DOM and getting them using their IDs, and `strongPassword` & `mediumPassword` for storing the Regex conditions.
+
+```JavaScript
+let timeout;
+let password = document.getElementById('0101')
+let strengthBadge = document.getElementById('0102')
+let strongPassword = new RegExp('(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[^A-Za-z0-9])(?=.{8,})')
+let mediumPassword = new RegExp('((?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[^A-Za-z0-9])(?=.{6,}))|((?=.*[a-z])(?=.*[A-Z])(?=.*[^A-Za-z0-9])(?=.{8,}))')
+```
+We then add an input event listener to the `password` input where we check the strength when the user has typed something, but we won't do it immediately after every input event. When typing quickly, we wait until a pause occurs so instead of immediately acting as the event handler, we set a timeout.
```

Comment:
```suggestion
Let's add an input event listener to the `password` input and check the strength when the user has typed something, but we won't do it immediately after every keystroke. When the user types quickly, we should wait until a pause occurs. So, instead of checking the strength immediately, we'll set a timeout.
```